test/functional/ex_cmds/cd_spec.lua

```diff
@@ -0,0 +1,99 @@
+-- Specs for :cd, :tcd, :lcd and hastabdir(), haslocaldir()
+
+local helpers = require('test.functional.helpers')
+local execute, eq, clear, eval, feed =
+  helpers.execute, helpers.eq, helpers.clear, helpers.eval, helpers.feed
+
+
+-- These directories will be created for testing
+local directories = {
+  'Xtest-functional-ex_cmds-cd_spec.1', -- Tab
+  'Xtest-functional-ex_cmds-cd_spec.2', -- Window
+  'Xtest-functional-ex_cmds-cd_spec.3', -- New global
+}
+
+-- Shorthand writing to get the current working directory
+local wd  = function() return eval('getcwd()'     ) end
+local htd = function() return eval('hastabdir()'  ) end
+local hld = function() return eval('haslocaldir()') end
+
+-- Test both the `cd` and `chdir` variants
+for _, cmd in ipairs {'cd', 'chdir'} do
+  describe(':*' .. cmd, function()
+    before_each(function()
+      clear()
+      for _, d in ipairs(directories) do
+        lfs.mkdir(d)
+      end
+    end)
+
+    after_each(function()
+      for _, d in ipairs(directories) do
+        lfs.rmdir(d)
+      end
+    end)
+
+    it('works', function()
```

Comment:
Would prefer if this test were broken into multiple small `it()` blocks instead of one giant block, unless that is painful for some reason.